### PR TITLE
Null safety for Shimmer effect

### PR DESCRIPTION
**Pull request by agavrel**
_Description of what this PR is changing or adding, and why:_
check that `context.findRenderObject() != null`

_Issues fixed by this PR (if any):_
prevent "type 'Null' is not a subtype of type 'RenderBox' in type cast" segmentation fault

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — CHANGES_REQUESTED** [review]
Please see https://github.com/dart-lang/site-shared/blob/main/doc/code-excerpts.md on how to update the code excerpts. Thanks!

File: src/cookbook/effects/shimmer-loading.md [hunk 1]
```diff
@@ -732,7 +732,8 @@ class ShimmerState extends State<Shimmer> with SingleTickerProviderStateMixin {
             _SlidingGradientTransform(slidePercent: _shimmerController.value),
       );
 
-  bool get isSized => (context.findRenderObject() as RenderBox).hasSize;
+  bool get isSized => context.findRenderObject() != null
```
Comment (navaronbracke) [inline src/cookbook/effects/shimmer-loading.md:735]:
```suggestion
  bool get isSized => (context.findRenderObject() as RenderBox?)?.hasSize ?? false;
```

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**